packages/asset-swapper/src/quote_consumers/exchange_proxy_swap_quote_consumer.ts

```diff
@@ -109,18 +107,46 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumerBase {
         }
 
         // This transformer will fill the quote.
-        transforms.push({
-            deploymentNonce: this.transformerNonces.fillQuoteTransformer,
-            data: encodeFillQuoteTransformerData({
-                sellToken,
-                buyToken,
-                side: isBuyQuote(quote) ? FillQuoteTransformerSide.Buy : FillQuoteTransformerSide.Sell,
-                fillAmount: isBuyQuote(quote) ? quote.makerAssetFillAmount : quote.takerAssetFillAmount,
-                maxOrderFillAmounts: [],
-                orders: quote.orders,
-                signatures: quote.orders.map(o => o.signature),
-            }),
-        });
+        if (quote.isTwoHop) {
+            const [firstHopOrder, secondHopOrder] = quote.orders;
+            transforms.push({
+                deploymentNonce: this.transformerNonces.fillQuoteTransformer,
+                data: encodeFillQuoteTransformerData({
+                    sellToken,
+                    buyToken: getTokenFromAssetData(firstHopOrder.makerAssetData),
```

Comment:
Are all of the following equivalent? If so might be worth just assigning it to `intermediateToken` and using it in the 3 places.
```typescript
getTokenFromAssetData(firstHopOrder.makerAssetData) 
=== getTokenFromAssetData(secondHopOrder.takerAssetData)
=== getTokenFromAssetData(quote.orders[0].makerAssetData)
```